railties/lib/rails/generators/rails/scaffold_controller/templates/api_controller.rb

```diff
@@ -0,0 +1,63 @@
+<% if namespaced? -%>
+require_dependency "<%= namespaced_file_path %>/application_controller"
+
+<% end -%>
+<% module_namespacing do -%>
+class <%= controller_class_name %>Controller < ApplicationController
+  before_action :set_<%= singular_table_name %>, only: [:show, :update, :destroy]
+
+  # GET <%= route_url %>
+  def index
+    @<%= plural_table_name %> = <%= orm_class.all(class_name) %>
+
+    render json: <%= "@#{plural_table_name}" %>
+  end
+
+  # GET <%= route_url %>/1
+  def show
+    render json: <%= "@#{singular_table_name}" %>
+  end
+
+  # POST <%= route_url %>
+  def create
+    @<%= singular_table_name %> = <%= orm_class.build(class_name, "#{singular_table_name}_params") %>
+
+    if @<%= orm_instance.save %>
+      render json: <%= "@#{singular_table_name}" %>, status: :created, location: <%= "@#{singular_table_name}" %>
+    else
+      render json: <%= "@#{orm_instance.errors}" %>, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT <%= route_url %>/1
+  def update
+    if @<%= orm_instance.update("#{singular_table_name}_params") %>
+      render json: <%= "@#{singular_table_name}" %>
+    else
+      render json: <%= "@#{orm_instance.errors}" %>, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE <%= route_url %>/1
+  def destroy
+    @<%= orm_instance.destroy %>
+
+    head :no_content
```

Comment:
Rails master implicitly renders [`head :no_content`](https://github.com/rails/rails/blob/master/actionpack/lib/action_controller/metal/implicit_render.rb#L14) if no template is found, so we don't need this line :smile: